streams/src/main/java/org/apache/kafka/streams/processor/internals/GlobalStreamThread.java

```diff
@@ -311,6 +317,22 @@ public void run() {
                 "Updating global state failed. You can restart KafkaStreams to recover from this error.",
                 recoverableException
             );
+        } catch (final Exception e) {
+            if (this.streamsUncaughtExceptionHandler == null) {
+                throw e;
+            }
+            if (Thread.getDefaultUncaughtExceptionHandler() != null && newHandler) {
+                log.error("Stream's new uncaught exception handler is set as well as the deprecated old handler." +
+                        "The old handler will be ignored as long as a new handler is set.");
```

Comment:
This doesn't look like an "error". At best it's a "warn" log, but only if we think that this combination definitely looks like a misconfiguration. Even then, why wouldn't we check for the misconfiguration in KafkaStreams, since both the new and old handlers would be set over there?

Reply:
I think it is simpler to check in the Stream thread because we don't in KafkaStreams if the handlers have been set so we would have to check the stream thread a global thread so it would be much easier to just check in the thread. I do agree that it should be bumped down to warn through.